### PR TITLE
Replace PullingMonitor timeout constants with config parameters

### PR DESCRIPTION
**Pull request by thescouser89**
This PR is to fix some of the issues in #1252

### CODE REVIEW

**Comment by thescouser89** [comment]
@jsenko I don't know what was up with the error in `PullingMonitorTest`. I formatted the pnc-config.json it was reading so that it used the new pnc-config format. I have no idea why we didn't hit that error before if the config format was wrong. Maybe the tests were disabled? No idea.

As for the integration tests, I added a default constructor so that JBoss could get its act together and do the injection properly.

I hope I didn't break anything! :speak_no_evil:

File: openshift-environment-driver/src/main/java/org/jboss/pnc/environment/monitor/PullingMonitor.java [hunk 1]
```diff
@@ -16,14 +16,20 @@
  * limitations under the License.
  */
 
-package org.jboss.pnc.common.monitor;
+package org.jboss.pnc.environment.monitor;
```
Comment (matejonnet) [inline openshift-environment-driver/src/main/java/org/jboss/pnc/environment/monitor/PullingMonitor.java]: Why moving this one to the environment module, can't we keep it generic ?

File: openshift-environment-driver/src/main/java/org/jboss/pnc/environment/monitor/PullingMonitor.java [hunk 2]
```diff
@@ -54,14 +54,23 @@
...
 
+    @Inject
+    private Configuration configuration;
+
     public PullingMonitor() {
+    }
+
+    @Inject
+    public PullingMonitor(Configuration configuration) {
+        this.configuration = configuration;
```
Comment (matejonnet) [inline openshift-environment-driver/src/main/java/org/jboss/pnc/environment/monitor/PullingMonitor.java]: There already is a `monitor`method that takes timeout as a parameter. There should be no need to make this class dependent on the configuration.

**Comment by jsenko** [comment]
The move is because of the circular dependency if common (where PullingMonitor was) would depend on configuration and vice versa. This in turn is because the monitor method is used 4 times, which would require the timeout configuration parameters to be passed each time. Instead they are supplied directly by the monitor. Is the move more problematic that this?

**Pull request closed without merging**